### ISSUE TITLE
reset the SlideshowLightbox

### ISSUE THREAD

**Issue by dgastudio** [silvia-odwyer/lightbox.js#26]
i'm using nextjs with lazy load and fetch to update the list of images
![CleanShot 2023-07-12 at 11 58 57@2x](https://github.com/silvia-odwyer/lightbox.js/assets/161524/b3e3a6f2-db32-4935-ab5c-11d99424d5c3)

the issue is then i'm adding new images to list, or reloading the list of existing ones, seems like SlideshowLightbox is being cached somehow.

is there an method to reset it?

**Comment by silvia-odwyer** [comment]
@dgastudio Hi Alexei,
One potential solution would be to pass the array of images to the `displayedImages` prop of the SlideshowLightbox component. This prop is only used to check if the lightbox needs to be updated, and isn't used to render the images in the lightbox. When there is a change in value detected, the component will update and add all the images with the data-lightboxjs attribute to the lightbox.

```jsx
<SlideshowLightbox displayedImages={images}>
  ...
</SlideshowLightbox>
```

Hopefully that helps, and let me know if you have any queries.

**Comment by dgastudio** [comment]
i'm pretty sure that i have seen somewhere reload function.
dear @silvia-odwyer with your suggestion, it will be not possible to make things with layout i'm having right now

**Comment by silvia-odwyer** [comment]
@dgastudio The `displayedImages` prop isn't used for rendering the images in the layout, it's only used to track changes in the array and then reload the lightbox.  So you could pass the data.items value, for instance, and when new images get added/removed, the lightbox will then update.

With regards a reload function, I'm just wondering how that would be added to the component, I could try to add this, but not quite sure what the preferred implementation would be.

**Comment by dgastudio** [comment]
@silvia-odwyer you are awesome, thank you very much!

**Comment by silvia-odwyer** [comment]
@dgastudio Just wanted to mention here that an alternative way of updating the lightbox has also been added. 

The library has been updated to include a `reset` function also. This involves creating a ref, passing this to the lightbox, and then the reset function can be called as a result.

Firstly, a ref would need to be created:
```js
let lightboxRef = createRef();
```

Then, just pass this ref to the SlideshowLightbox component:
```jsx
<SlideshowLightbox 
ref={lightboxRef}>
...
</SlideshowLightbox>
```

You can then call the reset function whenever you want to refresh the lightbox after new images have been added/removed/changed in the DOM. 

```js
lightboxRef.current.reset();
```

This would be useful to those who want to update the lightbox and aren't passing the images via an `images` prop. Hopefully this will help others in the future, and I'll update the documentation to include this too.

**Comment by dgastudio** [comment]
@silvia-odwyer 
awesome, thanks

**Comment by silvia-odwyer** [comment]
@kafejo Hi Aleš, 
The latest version (0.7.1) currently includes the reset function, is it not working for your use case? I can help with debugging the issue.

**Comment by kafejo** [comment]
I see, I have 0.7.1 and there's no `ref` on the `SlideshowLightbox`.

**Comment by silvia-odwyer** [comment]
@kafejo The SlideshowLightbox interface has been updated to also include the `ref`, it should now be available as part of the 0.7.2 version release ✅ If you have any other queries, just let me know, thanks!

**Comment by kafejo** [comment]
Great, thank you!
